lib_eio/path.mli

```diff
@@ -207,3 +207,17 @@ val rename : _ t -> _ t -> unit
 (** [rename old_t new_t] atomically unlinks [old_t] and links it as [new_t].
 
     If [new_t] already exists, it is atomically replaced. *)
+
+val symlink : ?exists_ok:bool -> link_to:string -> _ t -> unit
+(** [symlink ~link_to src] creates a symbolic link [src] to [link_to]. [src] is the
+    file that is created and [link_to] is the name used in the link.
+
+    The following program creates a new file called [./hello.txt] and then creates
+    a symbolic link called [./world.txt] that points to [./hello.txt].
+    {[
+      Eio.Path.(save ~create:(`If_missing 0o644) (env#cwd / "hello.txt")) "Hello World!";
+      Eio.Path.(symlink ~link_to:"hello.txt" (env#cwd / "world.txt"))
+    ]}
+
+    @param exist_ok If [false] (the default) then we raise {! Fs.Already_exists} if the symlink already exists with that target.
```

Comment:
What if it exists with a different target, or isn't a symlink? Also `exist` vs `exists`.

(could just drop the `exists_ok` feature and add it later)